### PR TITLE
Products: Variable product > Inventory section: hide stock status when "manage stock" is disabled

### LINKED ISSUE #2783: Products: Variable product > Inventory section shows (potentially inaccurate) stock status when "manage stock" is disabled
**Describe the bug**

As noted in https://github.com/woocommerce/woocommerce-ios/pull/2725, the Inventory section for a variable product should include a "Manage Stock" toggle but _not_ a Stock Status row.

However, the product details screen for a variable product still shows a stock status (In Stock/Out of Stock) when "Manage Stock" is toggled off at the product level. This could be ok if it matches the variation stock status, but updating the variation stock status doesn't necessarily update the product-level stock status shown in product details.

cc @jaclync 

**To Reproduce**

1. Go to Settings > Experimental Features and enable Product Editing.
2. Go to the Products tab and select a variable product.
3. Notice that under "Inventory" it shows a stock status (e.g. "In stock").
4. On the product details screen, tap "Inventory" to open the product's inventory settings.
5. Confirm "Manage stock" is toggled **off**.
6. Go back to product details and tap "Variations" to open the variations list.
7. Tap a variation to open its details.
8. Tap "Inventory" to open the variation's inventory settings.
9. Confirm "Manage stock" is toggled **off** and switch the stock status so it's opposite of the product stock status you noted in step 3 (e.g. if the product showed "In stock" switch it to "Out of stock").
10. Repeat steps 7-9 for every variation.
11. Go back to product details and notice that the stock status hasn't changed. (For example, even if every variation is "Out of stock" it still shows "In stock.") Notice that you can't pull to refresh that stock status.
12. Make some other change to the product details and tap "Update." Notice that this refreshes the product details and the product's Inventory section now shows the correct stock status.

**Screenshots**

Screencast: https://cloudup.com/cW90PqR4FKe

Stock status in product details:

**Expected behavior**

Since stock status can't be controlled at the product level, it would make sense not to show it there.

Alternately, if it does show a stock status, that status should update according to the stock status of the product variations (as soon as the variation stock status is updated).

**Mobile Environment**
Please include:
- Device: iPhone XS
- iOS version: 13.6.1
- WooCommerce iOS version: 5.0.0 TestFlight beta

### PR DESCRIPTION
**Pull request by jaclync**
Fixes #2783 

## Changes

- In the inventory row on the product details screen, the stock status is now only displayed when stock management is disabled & stock status is enabled (it's disabled for variable products)

### Variable product

- Make sure the products feature switch is on under Settings > Experimental Features
- Go to the Products tab
- Tap on a variable product whose stock management is disabled (inventory row > "Manage stock" toggle off) --> the stock status should not be shown in the inventory section anymore (it's shown on `develop`)

### Other product types

- Go to the Products tab
- Tap on a simple product whose stock management is disabled (inventory row > "Manage stock" toggle off) --> the stock status should be shown in the inventory section like before

## Example screenshots

Variable product - before | Variable product - after | Simple product
-- | -- | --
Simulator Screen Shot - iPhone 11 - 2020-09-09 at 14 50 41 | Simulator Screen Shot - iPhone 11 - 2020-09-09 at 14 54 42 | Simulator Screen Shot - iPhone 11 - 2020-09-09 at 14 54 58

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift [hunk 1]
```diff
@@ -199,7 +199,7 @@ private extension DefaultProductFormTableViewModel {
 
         if let stockQuantity = product.stockQuantity, product.manageStock {
             inventoryDetails.append(String.localizedStringWithFormat(Constants.stockQuantityFormat, stockQuantity))
-        } else if product.manageStock == false {
+        } else if product.manageStock == false && product.isStockStatusEnabled() {
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift:202]: Sanity check: can this be unit tested?

**Review by Ecarrion — APPROVED** [review]
Looks good!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift:202]: Good call! We didn't have existing unit tests, added two for simple and variable products when stock management is disabled: https://github.com/woocommerce/woocommerce-ios/pull/2786/commits/42f3c07c6deeee0ba6a81a400dedd53b5ed37ce3

**Review by Ecarrion — APPROVED** [review]
👏 👏

**Pull request merged**